### PR TITLE
Add CPU and RAM settings to VM resource

### PR DESCRIPTION
**Pull request by ja6a**
Adding the ability to set the CPU and RAM settings in the VM resource. The documentation describes the usage.

### CODE REVIEW

**Comment by ja6a** [comment]
**Note Regarding Force New**
Changing the RAM and CPU settings requires the VM to be shutdown. This is because the OS needs to boot up in order to process these settings. In terraform the standard approach of cloud providers when changing these resources is to recreate the VM resource, e.g., AWS, GCP, and Azure. Since this is an official provider we would like to stay with the best practise for these OS level changes. To clarify: the approach of changing the provider to stop and start the VM when these settings change - silently from the users point of view - will not be supported. The detailed reason is as follows: terraform has no knowledge of the application layer. It will not handle gracefully any application level considerations to perform a clean shutdown and restart.

**Comment by ja6a** [comment]
Refers to issue #6

File: skytap/resource_skytap_vm.go [hunk 1]
```diff
@@ -420,6 +437,54 @@ func updateVMResource(d *schema.ResourceData, meta interface{}, forceRunning boo
...
+		opts.Hardware.RAM = utils.Int(v.(int))
+	}
+
+	if opts.Hardware != nil {
+		vm, err := (*client).Get(*ctx, environmentID, id)
+		if err != nil {
+			return err
+		}
+		// check max cpu property
+		if opts.Hardware.CPUs != nil && *opts.Hardware.CPUs > *vm.Hardware.MaxCPUs {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go:465]: abstract the resource validation

File: skytap/resource_skytap_vm.go [hunk 2]
```diff
@@ -385,23 +404,21 @@ func updateVMResource(d *schema.ResourceData, meta interface{}, forceRunning boo
...
-	}
-	if v, ok := d.GetOk("name"); ok {
-		opts.Name = utils.String(v.(string))
-	}
-
-	log.Printf("[INFO] VM update options: %#v", spew.Sdump(opts))
-	vm, err := client.Update(ctx, environmentID, id, &opts)
-	if err != nil {
-		return fmt.Errorf("error updating vm (%s): %v", id, err)
+		log.Printf("[INFO] VM starting: %#v", spew.Sdump(opts))
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go:414]: Shall we move Sdump to a debug level

File: skytap/resource_skytap_vm.go [hunk 3]
```diff
@@ -420,6 +437,54 @@ func updateVMResource(d *schema.ResourceData, meta interface{}, forceRunning boo
 	return resourceSkytapVMRead(d, meta)
 }
 
+func updateVMResourceOptions(d *schema.ResourceData, client *skytap.VMsService, ctx *context.Context, environmentID string, id string) error {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go:440]: Rename to  configureVMResource, add documentation a comment why the configuration is necessary after creation,

**Review by pegerto — CHANGES_REQUESTED** [review]

**Comment by pegerto** [comment]
@cgriggs01  Hello Chris, can you review this PR? Thanks

File: skytap/resource_skytap_vm.go (hunk 2 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go:414]: this is planned for a future issue.

File: skytap/resource_skytap_vm.go (hunk 3 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go:440]: This recommendation was done in PR #13. The create and update flow have been clearly delineated.

File: skytap/resource_skytap_vm.go (hunk 1 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go:465]: This recommendation was done in PR #13.

File: skytap/resource_skytap_vm.go (hunk 2 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go:414]: see issue #18

**Review by pegerto — APPROVED** [review]

**Pull request merged**